wgengine/magicsock/magicsock.go

```diff
@@ -5105,6 +5126,82 @@ func (s derpAddrFamSelector) PreferIPv6() bool {
 	return false
 }
 
+type endpointTrackerEntry struct {
+	endpoint tailcfg.Endpoint
+	until    time.Time
+}
+
+type endpointTracker struct {
+	mu    sync.Mutex
+	cache map[netip.AddrPort]endpointTrackerEntry
+}
+
+const endpointTrackerLifetime = 5*time.Minute + 10*time.Second // slightly longer than a full netcheck period
```

Comment:
move this to the big const(...) block of magic time durations elsewhere in this file/package?